templates/taxi/manufacturer_list.html

```diff
@@ -1,33 +1,39 @@
 {% extends "base.html" %}
 
 {% block content %}
-    <h1>Manufacturer List
-    </h1>
+  <h1>Manufacturer List</h1>
+  <a href="{% url 'taxi:manufacturer-create' %}" class="btn btn-primary">Add manufacturer</a>
+  {% if manufacturer_list %}
+    <table class="table">
+      <tr>
+        <th>ID</th>
+        <th>Name</th>
+        <th>Country</th>
+        <th>Upgrade</th>
+        <th>Delete</th>
+      </tr>
 
-    {% if manufacturer_list %}
-      <table class="table">
+      {% for manufacturer in manufacturer_list %}
         <tr>
-          <th>ID</th>
-          <th>Name</th>
-          <th>Country</th>
+          <td>
+              {{ manufacturer.id }}
+          </td>
+          <td>
+              {{ manufacturer.name }}
+          </td>
+          <td>
+              {{ manufacturer.country }}
+          </td>
+          <td>
+              <a href="{% url 'taxi:manufacturer-update' pk=manufacturer.id%}">Upgrade</a>
```

Comment:
Consider adding a space before the closing `%}` in the URL tag for consistency: `{% url 'taxi:manufacturer-update' pk=manufacturer.id %}`.